test/e2e/upgrade/upgrade.go

```diff
@@ -625,6 +626,21 @@ func clusterUpgrade(f *framework.Framework, c configv1client.Interface, dc dynam
 		func() (error, bool) {
 			framework.Logf("Waiting on pools to be upgraded")
 			if err := wait.PollImmediate(10*time.Second, 30*time.Minute, func() (bool, error) {
+
+				nodes, err := kubeClient.CoreV1().Nodes().List(context.Background(), metav1.ListOptions{})
```

Comment:
It's hard to imagine a MachineConfigPool update that cordons and drains control-plane nodes where we never see a single `master` Node that's `Unschedulable=True` in this 10s poll loop, so looks good to me.